src/insight/insightClient.ts

```diff
@@ -213,6 +219,14 @@ export class CoveoInsightClient {
         return this.logCustomEvent(InsightEvents.expandToFullUI, metadataToSend);
     }
 
+    public logDocumentOpen(info: PartialDocumentInformation, identifier: DocumentIdentifier, metadata?: CaseMetadata) {
+        if (metadata) {
+            const metadataToSend = generateMetadataToSend(metadata, false);
+            return this.logClickEvent(SearchPageEvents.documentOpen, info, identifier, metadataToSend);
+        }
+        return this.logClickEvent(SearchPageEvents.documentOpen, info, identifier);
+    }
```

Comment:
We can avoid repeating the `logClickEvent` arguments by using a ternary. IMO it is a little more readable.

```suggestion
        return this.logClickEvent(
          SearchPageEvents.documentOpen,
          info,
          identifier,
          metadata ? generateMetadataToSend(metadata, false) : undefined);
    }
```